src/components/forms/NetworkDevicesForm/InheritedNetworkRow.tsx

```diff
@@ -0,0 +1,146 @@
+import ResourceLink from "components/ResourceLink";
+import { getConfigurationRowBase } from "components/ConfigurationRow";
+import ReadOnlyAclsList from "components/forms/NetworkDevicesForm/ReadOnlyAclsList";
+import type { InstanceAndProfileFormikProps } from "components/forms/instanceAndProfileFormValues";
+import type { InheritedNetwork } from "util/configInheritance";
+import type { LxdNetwork } from "types/network";
+import type { MainTableRow } from "@canonical/react-components/dist/components/MainTable/MainTable";
+import type { EditInstanceFormValues } from "pages/instances/EditInstance";
+import classNames from "classnames";
+import NetworkDevice from "./NetworkDevice";
+import type { LxdNicDevice, LxdNoneDevice } from "types/device";
+import { ensureEditMode } from "util/instanceEdit";
+import NetworkDeviceActionButtons from "./NetworkDeviceActionButtons";
+import { addDevice, addNoneDevice } from "util/formDevices";
+
+interface Props {
+  device: InheritedNetwork;
+  project: string;
+  managedNetworks: LxdNetwork[];
+  formik: InstanceAndProfileFormikProps;
+  focusNetwork: (id: number) => void;
+  removeNetwork: (deviceName: string) => void;
+  isOverridden: boolean;
+}
+
+export const getInheritedNetworkRow = ({
+  device,
+  project,
+  managedNetworks,
+  formik,
+  focusNetwork,
+  removeNetwork,
+  isOverridden: isOverridden,
+}: Props): MainTableRow => {
+  const readOnly = (formik.values as EditInstanceFormValues).readOnly;
+  const overrideDevice = formik.values.devices.find(
+    (t) => t.name === device.key,
+  ) as LxdNicDevice | LxdNoneDevice;
+
+  const overrideNetwork = managedNetworks.find(
+    (t) =>
+      t.name ===
+      (formik.values.devices.find((t) => t.name === device.key) as LxdNicDevice)
+        ?.network,
```

Comment:
```suggestion
  const overrideNetwork = managedNetworks.find(
    (t) =>
      t.name === overrideDevice?.network,
```
I think we can use the variable from above here to simplify.